FirstVersionsofDiTauHLTPaths_forPhase2 (master)

hsert opened the pull request:
#### PR description:

This PR adds the first version of di-tau Phase2 HLT paths with charged and DeepTau isolation. The performance of triggers is not optimal yet. The current performance and also details for the paths can be seen in the following slides:

https://indico.cern.ch/event/1306042/contributions/5492549/attachments/2686771/4661683/Phase2_TauTriggerMeeting_hsert.pdf


#### PR validation:

The basic tests has been performed, suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)
- "scram b runtests" command gives the following output:
  Creating test log file logs/slc7_amd64_gcc11/testing.log
  Fail  115s ... HLTrigger/Configuration/testAccessToEDMInputsOfHLTTests
  >> Test sequence completed for CMSSW CMSSW_13_1_0_pre3
- runTheMatrix test workflow seems to be working fine.

The checks are performed with CMSSW release of CMSSW_13_1_0_pre3, so it would be good to have it backported to this release if possible.

hsert committed:
First versions of ditau HLT paths for Phase2 with charged isolation and DeepTau Isolation (using Run2 training) are added.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36577

- This PR adds an extra 52KB to repository

- Found files with invalid states:
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsDoublePFTau40TrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsSelectedPFTausTrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsL1JetsHLTDoublePFTauTrackPt1MediumChargedIsolationMatch_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9

cmsbuild commented:
A new Pull Request was created by  @hsert (HaleSert) for master.

It involves the following packages:

- HLTrigger/Configuration (**hlt**)


@cmsbuild, @missirol, @Martin-Grunewald can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @missirol, @silviodonato, @SohamBhattacharya, @beaucero this is something you requested to watch as well.
@perrotta, @dpiparo, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

missirol commented on HLTrigger/Configuration/python/HLT_75e33_cff.py:
```diff
@@ -283,6 +290,12 @@
     fragment.HLT_DoubleEle23_12_Iso_L1Seeded,
     fragment.HLT_Diphoton30_23_IsoCaloId_L1Seeded,
 
+    #AC
+    #fragment.HLT_DoublePFTau,
```
What is "AC" ? Can you please remove commented code if not needed ?

missirol commented on HLTrigger/Configuration/python/HLT_75e33_cff.py:
```diff
@@ -283,6 +290,12 @@
 
+    #AC
+    #fragment.HLT_DoublePFTau,
+    fragment.HLT_DoublePFTauHPS,
+    fragment.HLT_DoubleMediumChargedIsoPFTauHPS,
+    fragment.HLT_DoubleMediumDeepTauPFTauHPS,
```
Shouldn't the Path names contain some info on which thresholds are applied ?

missirol commented on HLTrigger/Configuration/python/HLT_75e33/paths/HLT_DoubleMediumChargedIsoPFTauHPS_cfi.py:
```diff
@@ -0,0 +1,26 @@
+    HLTHPSMediumChargedIsoPFTauSequence +
+    hltHpsSelectedPFTausTrackPt1MediumChargedIsolation +
+    hltHpsDoublePFTau40TrackPt1MediumChargedIsolation +
+    #hltHpsL1JetsHLTDoublePFTauTrackPt1MediumChargedIsolationMatch +
```
Please remove commented code if not needed.

missirol commented:
@hsert , please improve the title of the PR with something more descriptive.

Will this PR need to be backported to any earlier release cycles ?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36578

- This PR adds an extra 76KB to repository

- Found files with invalid states:
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsDoublePFTau40TrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsSelectedPFTausTrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsL1JetsHLTDoublePFTauTrackPt1MediumChargedIsolationMatch_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9

missirol commented:
Waiting for https://github.com/cms-sw/cmssw/pull/42562#discussion_r1293134509 and https://github.com/cms-sw/cmssw/pull/42562#issuecomment-1676884084 to be addressed before testing this PR.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36582

- This PR adds an extra 104KB to repository

- Found files with invalid states:
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsDoublePFTau40TrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/paths/HLT_DoublePFTauHPS_cfi.py:
    - Added: 19f5eb4f7269c7573c61ebeccdaedbc4529125df
    - Deleted: 161c559130d486a4faf44ea912045a4a206104fb
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltPFTauHPS_cfi.py:
    - Added: 19f5eb4f7269c7573c61ebeccdaedbc4529125df
    - Modified: 0999ba29015aba7588181719d67f5886c36bb2bd
    - Deleted: 161c559130d486a4faf44ea912045a4a206104fb
  - HLTrigger/Configuration/python/HLT_75e33/paths/HLT_DoubleMediumChargedIsoPFTauHPS_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Modified: 57dff450506c973a4525893292d16ebeeac289e9, 0999ba29015aba7588181719d67f5886c36bb2bd
    - Deleted: 161c559130d486a4faf44ea912045a4a206104fb
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsSelectedPFTausTrackPt1MediumChargedIsolation_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9
  - HLTrigger/Configuration/python/HLT_75e33/paths/HLT_DoubleMediumDeepTauPFTauHPS_cfi.py:
    - Added: e87c4a4d136911d95e6a77eea1c532bca4c8f369
    - Deleted: 161c559130d486a4faf44ea912045a4a206104fb
  - HLTrigger/Configuration/python/HLT_75e33/sequences/hltHpsL1JetsHLTDoublePFTauTrackPt1MediumChargedIsolationMatch_cfi.py:
    - Added: 93becf79fd2b4c8d96cf68c5fadc9a9ed45ca0f8
    - Deleted: 57dff450506c973a4525893292d16ebeeac289e9

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

hsert commented:
> @hsert , please improve the title of the PR with something more descriptive.
> 
> Will this PR need to be backported to any earlier release cycles ?

@missirol I modified the title and comment part to answer your questions here. I hope it is fine now.

missirol commented:
> Found files with invalid states:

Please merge (or "squash") the 6 commits into 1. After that, we can try to test this PR.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsSelectedPFTausTrackPt1MediumChargedIsolation_cfi.py:
```diff
@@ -0,0 +1,16 @@
+    src = cms.InputTag( "hltHpsPFTauProducer" ),
+    cut = cms.string( "pt > 0" ),
+    discriminators = cms.VPSet( 
+      #cms.PSet(  discriminator = cms.InputTag( "hltHpsPFTauTrackPt1Discriminator" ),
+        #selectionCut = cms.double( 0.01 )
+      #),
```
Still some commented code. Remove if possible.

hsert replied:
I kept it on purpose, since it should be probably there based on Run2 paths but currently that part gives an error. We should still understand it. Of course we can remove it and once we fix it, we can add it later. If you suggest to do so, I can do it.

missirol replied:
Since there is a reason to possibly keep it, I don't have a strong preference. I let you decide.

hsert replied:
ok, then I prefer to keep it just to be on the safe side that the information won't be lost.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/modules/hltPreDoublePFTauHPS_cfi.py:
```diff
@@ -0,0 +1,7 @@
+#AC
```
Remove.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/paths/HLT_DoubleMediumDeepTauPFTauHPS35_eta2p1_cfi.py:
```diff
@@ -0,0 +1,26 @@
+from ..modules.hltHpsSelectedPFTausMediumDitauWPDeepTau_cfi import *
+from ..modules.hltHpsDoublePFTau35MediumDitauWPDeepTau_cfi import *
+
+
```
Remove extra space.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTHPSDeepTauPFTauSequence_cfi.py:
```diff
@@ -0,0 +1,26 @@
+    hltHpsPFTauBasicDiscriminatorsdR03ForDeepTau + 
+    hltHpsPFTauDeepTauProducer 
+)
+
+
+
+
```
Remove extra spaces.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTHPSDeepTauPFTauSequence_cfi.py:
```diff
@@ -0,0 +1,26 @@
+from ..modules.hltHpsPFTauBasicDiscriminatorsdR03ForDeepTau_cfi import *
+from ..modules.hltHpsPFTauDeepTauProducer_cfi import *
+
+
```
Remove extra space.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTHPSMediumChargedIsoPFTauSequence_cfi.py:
```diff
@@ -0,0 +1,13 @@
+from ..modules.hltHpsPFTauMediumRelativeChargedIsolationDiscriminator_cfi import *
+from ..modules.hltHpsPFTauMediumAbsOrRelChargedIsolationDiscriminator_cfi import *
+
+
```
Remove extra space.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTHPSMediumChargedIsoPFTauSequence_cfi.py:
```diff
@@ -0,0 +1,13 @@
+    hltHpsPFTauMediumRelativeChargedIsolationDiscriminator + 
+    hltHpsPFTauMediumAbsOrRelChargedIsolationDiscriminator 
+)
+
```
Remove extra space.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTPFTauHPS_cfi.py:
```diff
@@ -0,0 +1,28 @@
+
```
Remove extra space.

missirol commented on HLTrigger/Configuration/python/HLT_75e33/sequences/HLTPFTauHPS_cfi.py:
```diff
@@ -0,0 +1,28 @@
+from ..modules.hltHpsPFTauTrack_cfi import *
+
+HLTPFTauHPS = cms.Sequence(
+hltTauPFJets08Region +
+hltHpsTauPFJetsRecoTauChargedHadronsWithNeutrals + 
+hltPFTauPiZeros + 
+hltHpsCombinatoricRecoTaus + 
+hltHpsSelectionDiscriminator + 
+hltHpsPFTauProducerSansRefs + 
+hltHpsPFTauProducer + 
+hltHpsPFTauDiscriminationByDecayModeFindingNewDMs + 
+hltHpsPFTauTrackFindingDiscriminator + 
+hltHpsSelectedPFTausTrackFinding + 
+hltHpsPFTauTrack 
```
Please indent.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36585

- This PR adds an extra 48KB to repository

missirol commented on HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsCombinatoricRecoTaus_cfi.py:
```diff
@@ -0,0 +1,116 @@
+    jetSrc = cms.InputTag( "hltAK4PFJets" ),
+    builders = cms.VPSet( 
+      cms.PSet(  decayModes = cms.VPSet( 
+  cms.PSet(  maxPiZeros = cms.uint32( 0 ),
+    maxTracks = cms.uint32( 6 ),
+    nPiZeros = cms.uint32( 0 ),
+    nCharged = cms.uint32( 1 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 6 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 1 ),
+    nPiZeros = cms.uint32( 1 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 5 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 1 ),
+    nPiZeros = cms.uint32( 2 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 0 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 2 ),
+    nPiZeros = cms.uint32( 0 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 3 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 2 ),
+    nPiZeros = cms.uint32( 1 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 0 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 3 ),
+    nPiZeros = cms.uint32( 0 )
+  ),
+  cms.PSet(  maxPiZeros = cms.uint32( 3 ),
+    maxTracks = cms.uint32( 6 ),
+    nCharged = cms.uint32( 3 ),
+    nPiZeros = cms.uint32( 1 )
+  )
+),
+        isolationConeSize = cms.double( 0.5 ),
```
Please fix the indentation of these parameters.

missirol reviewed (commented):
I left some comments on cosmetics inline. Please squash again once done (so 1 commit, ultimately).

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36586

- This PR adds an extra 48KB to repository

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36587

- This PR adds an extra 48KB to repository

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/summary.html
**COMMIT**: 487f4f3d843558479eef08627f8499fd06cbfd3b
**CMSSW**: CMSSW_13_3_X_2023-08-14-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42562/34284/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
```
----- Begin Fatal Exception 15-Aug-2023 11:46:54 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'HLT_DoubleMediumDeepTauPFTauHPS35_eta2p1'
   [2] Calling method for module RecoTauPiZeroProducer/'hltPFTauPiZeros'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Vertex>
Looking for module label: pixelVertices
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
```
----- Begin Fatal Exception 15-Aug-2023 11:47:09 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'HLT_DoubleMediumDeepTauPFTauHPS35_eta2p1'
   [2] Calling method for module RecoTauPiZeroProducer/'hltPFTauPiZeros'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Vertex>
Looking for module label: pixelVertices
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
```
----- Begin Fatal Exception 15-Aug-2023 11:50:31 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'HLT_DoubleMediumDeepTauPFTauHPS35_eta2p1'
   [2] Calling method for module RecoTauPiZeroProducer/'hltPFTauPiZeros'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Vertex>
Looking for module label: pixelVertices
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [25034.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrix-results/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
</details>



## RelVals-INPUT

- [20834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/20834.0_TTbar_14TeV+2026D88)```20834.0_TTbar_14TeV+2026D88/step2_TTbar_14TeV+2026D88.log```
- [20834.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/20834.501_TTbar_14TeV+2026D88_Patatrack_PixelOnlyCPU)```20834.501_TTbar_14TeV+2026D88_Patatrack_PixelOnlyCPU/step2_TTbar_14TeV+2026D88_Patatrack_PixelOnlyCPU.log```
- [20834.502](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/20834.502_TTbar_14TeV+2026D88_Patatrack_PixelOnlyGPU)```20834.502_TTbar_14TeV+2026D88_Patatrack_PixelOnlyGPU/step2_TTbar_14TeV+2026D88_Patatrack_PixelOnlyGPU.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [23634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/23634.0_TTbar_14TeV+2026D95)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24834.0_TTbar_14TeV+2026D98)
- [24834.103](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24834.103_TTbar_14TeV+2026D98Aging3000)
- [24834.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24834.5_TTbar_14TeV+2026D98_pixelTrackingOnly)
- [24834.9](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24834.9_TTbar_14TeV+2026D98_vectorHits)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [25034.114](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34284/runTheMatrixINPUT-results/25034.114_TTbar_14TeV+2026D98PU_OTInefficiency10PC)
</details>

missirol commented:
@hsert , looks like this does not take into account the changes introduced in #42491. If so, please update this PR accordingly.

hsert commented:
> @hsert , looks like this does not take into account the changes introduced in #42491. If so, please update this PR accordingly.

@missirol, has that PR already merged to previous CMSSW releases? I tried both CMSSW_13_3_0_pre1 and CMSSW_13_2_0 releases, and couldn't reproduce the error that you are receiving due to the mentioned PR. In short, in which release I should test it. Thanks in advance.

missirol commented:
I think you need a so-called Integration Build (IB) of CMSSW, e.g. `CMSSW_13_3_X_2023-08-15-2300` (just use it as any other release).

hsert commented:
> I think you need a so-called Integration Build (IB) of CMSSW, e.g. `CMSSW_13_3_X_2023-08-15-2300` (just use it as any other release).

I followed the following steps to test it, and I still did not receive any complains about the name of the module labels. 

cmsrel CMSSW_13_3_X_2023-08-15-2300
cd CMSSW_13_3_X_2023-08-15-2300/src
cmsenv
git cms-init
git cms-merge-topic hsert:tauHLTPhase2_dev
scram b 

cmsDriver.py Phase2 -s HLT:75e33 --processName=HLTX --conditions auto:phase2_realistic_T21 --geometry Extended2026D88 --era Phase2C17I13M9 --eventcontent FEVTDEBUGHLT --customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_1000 --filein=/store/relval/CMSSW_13_3_0_pre1/RelValTTbar_14TeV/GEN-SIM-DIGI-RAW/131X_mcRun4_realistic_v6_2026D98noPU-v1/2580000/1b05411f-1514-43ed-bb8a-651523df42ca.root -n 10 --nThreads 1 --no_exec

cmsRun Phase2_HLT.py

Is there anything that I am missing? 

In addition, in the beginning of the error message I have seen this message "User test area: For local testing, you can use /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42562/34284/install.sh to create a dev area with all the needed externals and cmssw changes." and sourced that script to create an environment, but then I received errors about GLIBC like "/lib64/libc.so.6: version `GLIBC_2.28' not found"... So, I couldn't test it in this env.

If you have any idea to help me here, that would be great.

perrotta commented:
@hsert I don't know what the cmsDriver command you run is intended for, but if you run in your setup any of the failing workflows, e.g.
```
runTheMatrix.py -l 24896.0
```
you'll get exactly the same errors as in these bot tests

missirol commented:
> I followed the following steps to test it, and I still did not receive any complains about the name of the module labels.
>
>[..]
>
> cmsDriver.py Phase2 -s HLT:75e33 --processName=HLTX --conditions auto:phase2_realistic_T21 --geometry Extended2026D88 --era Phase2C17I13M9 --eventcontent FEVTDEBUGHLT --customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_1000 --filein=/store/relval/CMSSW_13_3_0_pre1/RelValTTbar_14TeV/GEN-SIM-DIGI-RAW/131X_mcRun4_realistic_v6_2026D98noPU-v1/2580000/1b05411f-1514-43ed-bb8a-651523df42ca.root -n 10 --nThreads 1 --no_exec

This might be the case if that input file already contains the input collections in question (if they are not in the configuration used for the job, `cmsRun` can fall back on the collections available in the EDM input file). Try, for example, `runTheMatrix.py -l 24896.0`: that should fail in the area built with your recipe.

perrotta commented:
With #42491:
- `pixelVertices` was renamed to `hltPhase2PixelVertices`

Please do the same here

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36608

- This PR adds an extra 48KB to repository

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

hsert commented:
@missirol and @perrotta thanks for your comments, with the suggested changes the runtheMatrix.py test is working for me. I hope it is fine now (I merged the commits again).

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36609

- This PR adds an extra 36KB to repository

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

perrotta commented:
please test

missirol commented on HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsPFTauBasicDiscriminatorsForDeepTau_cfi.py:
```diff
@@ -0,0 +1,106 @@
+        minTrackHits = cms.uint32( 3 ),
+        minGammaEt = cms.double( 1.0 )
+      ),
+      primaryVertexSrc = cms.InputTag( "hltPhase2PixelVerticess" ),
```
```suggestion
      primaryVertexSrc = cms.InputTag( "hltPhase2PixelVertices" ),
```

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42562/36615

- This PR adds an extra 36KB to repository

cmsbuild commented:
Pull request #42562 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented:
@hsert , have you tested the 3 failing wfs locally (after the typo fix) ?

missirol commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ede46d/34324/summary.html
**COMMIT**: e2c2781fe5de5f9280d0f8b0f3aca6f5d2df7267
**CMSSW**: CMSSW_13_3_X_2023-08-16-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42562/34324/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially added 178 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 72 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3152915
 - DQMHistoTests: Total failures: 274
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3152619
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-16-1100+ede46d/58465/triggerResults): no differences found

missirol commented:
@SohamBhattacharya @rovere , do you have any comments on this PR ? See also the question in https://github.com/cms-sw/cmssw/pull/42562#issuecomment-1676884084 (I guess there should be backports down to 13_1_X).

> Will this PR need to be backported to any earlier release cycles ?

missirol commented:
> You potentially added 178 lines to the logs

https://github.com/cms-sw/cmssw/pull/42562#issuecomment-1681201183 (with its spurious log messages and DQM differences) looks very similar to https://github.com/cms-sw/cmssw/pull/42497#issuecomment-1670609280, so another example of #42525, unrelated to this PR.

rovere commented:
Ciao @missirol I do not have any additional comments on top of what has been already requested and implemented.
I'd be in favour of having a backport to 13_1.

missirol commented:
+hlt

 - Spurious DQM differences, see https://github.com/cms-sw/cmssw/pull/42562#issuecomment-1681242888.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @antoniovilela, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
+1

hsert commented:
> 



> > Ciao @missirol I do not have any additional comments on top of what has been already requested and implemented. I'd be in favour of having a backport to 13_1.
> 
> Okay, thanks, Marco.
> 
> @hsert , can you please open PRs to backport this update to `13_2_X` and `13_1_X` ?

@missirol, sure I will do it. Just to clarify before proceed, I am using CMSSW_13_1_X_2023-08-18-1100 and CMSSW_13_2_X_2023-08-18-1100  releases for testing. After initialization, I rebased the branch with 
```git cms-rebase-topic --old-base CMSSW_13_3_X_2023-08-15-1100 hsert:tauHLTPhase2_dev```
and now running the runMatrix test by using the following command
```runTheMatrix.py -l limited -i all --ibeos```

Once the tests are completed without an issue, I will do PRs for each releases separately.

If there is anything that I am missing, please let me know.

missirol commented:
@hsert , what is the status of the backport PRs ?
